src/features/playground/shareLinks/encoder/URLEncoder.tsx

```diff
@@ -0,0 +1,53 @@
+import { FSModule } from 'browserfs/dist/node/core/FS';
+import { compressToEncodedURIComponent } from 'lz-string';
+import qs from 'query-string';
+import { useState } from 'react';
+import { retrieveFilesInWorkspaceAsRecord } from 'src/commons/fileSystem/utils';
+import { useTypedSelector } from 'src/commons/utils/Hooks';
+import { EditorTabState } from 'src/commons/workspace/WorkspaceTypes';
+
+import ShareLinkState from '../ShareLinkState';
+
+export const useURLEncoder = () => {
+  const isFolderModeEnabled = useTypedSelector(
+    state => state.workspaces.playground.isFolderModeEnabled
+  );
+
+  const editorTabs = useTypedSelector(state => state.workspaces.playground.editorTabs);
+  const editorTabFilePaths = editorTabs
+    .map((editorTab: EditorTabState) => editorTab.filePath)
+    .filter((filePath): filePath is string => filePath !== undefined);
+  const activeEditorTabIndex: number | null = useTypedSelector(
+    state => state.workspaces.playground.activeEditorTabIndex
+  );
+  const chapter = useTypedSelector(state => state.workspaces.playground.context.chapter);
+  const variant = useTypedSelector(state => state.workspaces.playground.context.variant);
+  const execTime = useTypedSelector(state => state.workspaces.playground.execTime);
+  const fileSystem = useGetFileSystem();
+
+  const result: Partial<ShareLinkState> = {
```

Comment:
Ditto on typing, should be `ShareLinkState` where the share link state itself is a partial type.

Reply:
Changed as referred above.